lib/plugin.js

```diff
@@ -71,9 +72,15 @@ ManifestPlugin.prototype.apply = function(compiler) {
       return memo;
     }, {}));
 
-    // Append optional basepath onto all references.
-    // This allows output path to be reflected in the manifest.
-    if (this.opts.basePath) {
+    if (this.opts.useBasePathAndPublicPath) {
```

Comment:
This seems klunky. These path components aren't mutually exclusive, so they should be able to be applied sequentially. Also in this `useBasePathAndPublicPath` block, the `basePath` setting is not consistent with the way it works by itself below.

So, I'd propose gitting rid of `useBasePathAndPublicPath`, break the `if/else` of base and public paths apart into individual `if`'s, then reverse the order so that we apply public path first (if defined) to values, and then prepend base path (if defined) afterward to all keys and all values.